### PR TITLE
Check language files and database schema with app code checker

### PR DESCRIPTION
**Pull request by nickvergessen**
- [x] Database problem in 2FA backup codes https://github.com/nextcloud/server/pull/4425

Brings #4485 to apps

### CODE REVIEW

File: autotest-checkers.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        [ "$app" == "testing" ] || \
+        [ "$app" == "twofactor_backupcodes" ] || \
+        [ "$app" == "updatenotification" ] || \
+        [ "$app" == "user_ldap" ]
+    then
+        ./occ app:check-code -c strong-comparison "$app"
+    else
+        ./occ app:check-code "$app"
+    fi
+    RESULT=$?
```
Comment (nickvergessen) [inline autotest-checkers.sh]: Not sure this exits correctly, I guess we need to merge the result codes?

Comment (rullzer) [inline autotest-checkers.sh]: It will just take the last code

File: .drone.yml [hunk 2]
```diff
@@ -15,18 +15,7 @@ pipeline:
...
-      - php ./build/translation-checker.php
-      - php ./build/htaccess-checker.php
-      - ./occ app:check-code admin_audit
-      - ./occ app:check-code comments
-      - ./occ app:check-code federation
-      - ./occ app:check-code sharebymail
-      - ./occ app:check-code systemtags
-      - ./occ app:check-code theming
-      - ./occ app:check-code workflowengine
+      - ./autotest-checkers.sh
```
Comment (rullzer) [inline .drone.yml]: Why a single job?

File: autotest-checkers.sh (hunk 1 above)
Comment (rullzer) [inline autotest-checkers.sh]: RESULT=$(($RESULT+$?))

Just do that below each one... and return that at the end ;)

**Review by rullzer — APPROVED** [review]

File: .drone.yml (hunk 2 above)
Comment (nickvergessen) [inline .drone.yml]: it was before, but now devs can run this job in a simple fashion aswell

File: autotest-checkers.sh (hunk 1 above)
Comment (nickvergessen) [inline autotest-checkers.sh]: Fixed, thanks

**Comment by nickvergessen** [comment]
Rebased and ready to review

File: autotest-checkers.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        [ "$app" == "testing" ] || \
+        [ "$app" == "twofactor_backupcodes" ] || \
+        [ "$app" == "updatenotification" ] || \
+        [ "$app" == "user_ldap" ]
+    then
+        ./occ app:check-code --skip-checkers "$app"
+    else
+        ./occ app:check-code "$app"
+    fi
+    RESULT=$(($RESULT+$?))
```
Comment (MorrisJobke) [inline autotest-checkers.sh]: This is not really helpful, because the log doesn't show what is the cause for the failure 🙈

**Comment by MorrisJobke** [comment]
The file runs fine locally. :/

File: core/Command/App/CheckCode.php [hunk 4]
```diff
@@ -70,6 +72,12 @@ protected function configure() {
 				[ 'private', 'deprecation', 'strong-comparison' ]
 			)
 			->addOption(
+				'--skip-checkers',
```
Comment (MorrisJobke) [inline core/Command/App/CheckCode.php]: Nice that you added this, but it isn't evaluated in the code below 🙈

Comment (MorrisJobke) [inline core/Command/App/CheckCode.php]: I fixed this.

**Review by MorrisJobke — APPROVED** [review]

File: autotest-checkers.sh [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+RESULT=$(($RESULT+$?))
+bash ./build/mergejschecker.sh
+RESULT=$(($RESULT+$?))
+php ./build/translation-checker.php
+RESULT=$(($RESULT+$?))
+php ./build/htaccess-checker.php
+RESULT=$(($RESULT+$?))
+
+
+for app in $(find "apps/" -mindepth 1 -maxdepth 1 -type d -exec basename {} \;); do
```
Comment (MorrisJobke) [inline autotest-checkers.sh]: And I made this compatible with the UNIX find (the previous was only for GNU find which didn't worked on UNIX(

**Comment by LukasReschke** [comment]
This broke the updater… The release script has to remove that file from the built… I will adjust but all users on existing betas will have to manually delete this file…

**Comment by nickvergessen** [comment]
Sorry :/ awkward

**Pull request merged**